datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java

```diff
@@ -68,9 +70,22 @@ protected Selectable doAccept(HandlerEvent<DatalakeFullBackupWaitRequest> event)
             LOGGER.info("Start polling datalake full backup status for id: {}", sdxId);
             PollingConfig pollingConfig = new PollingConfig(sleepTimeInSec, TimeUnit.SECONDS, duration,
                     TimeUnit.MINUTES);
-            sdxBackupRestoreService.waitForDatalakeDrBackupToComplete(sdxId, request.getOperationId(), request.getUserId(),
+            DatalakeBackupStatusResponse backupStatusResponse = sdxBackupRestoreService.waitForDatalakeDrBackupToComplete(
+                    sdxId, request.getOperationId(), request.getUserId(),
                     pollingConfig, "Full backup");
-            response = new DatalakeBackupSuccessEvent(sdxId, userId, request.getOperationId());
+            switch (backupStatusResponse.getState()) {
+                case CANCELLED:
+                    response = new DatalakeBackupCancelledEvent(sdxId, userId, request.getOperationId());
+                    break;
+                case SUCCESSFUL:
+                case VALIDATION_SUCCESSFUL:
+                    response = new DatalakeBackupSuccessEvent(sdxId, userId, request.getOperationId());
+                    break;
+                // failed states are handled via Exception catch block
+                default:
+                    LOGGER.warn("Unknown datalake backup status, assuming successful backup: {}", backupStatusResponse.getState());
```

Comment:
the log statement contains the state - I don't understand what more we can log?
 Even more - VALIDATION_FAILED amd FAILED will never end up here because they will be thrown by `waitForDatalakeDrBackupToComplete` . There is an obscure code there but it was there and it is still there.